poke: 2.3 -> 2.4

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for poke is: "Interactive, extensible editor for binary data"

meta.homepage for poke is: "http://www.jemarch.net/poke"

meta.changelog for poke is: "https://git.savannah.gnu.org/cgit/poke.git/plain/ChangeLog?h=releases/poke-2.4"


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 2.4 with grep in /nix/store/q3s8312093kwqnh08sr9zcr8f22hh9xh-poke-2.4
- found 2.4 in filename of file in /nix/store/q3s8312093kwqnh08sr9zcr8f22hh9xh-poke-2.4

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

1 package rebuild(s)

1 x86_64-linux rebuild(s)
1 x86_64-darwin rebuild(s)


First fifty rebuilds by attrpath
poke
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/q3s8312093kwqnh08sr9zcr8f22hh9xh-poke-2.4 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A poke https://github.com/r-ryantm/nixpkgs/archive/6810e93f132ad45a3e3501c8eec0f5978bcad82e.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/q3s8312093kwqnh08sr9zcr8f22hh9xh-poke-2.4
ls -la /nix/store/q3s8312093kwqnh08sr9zcr8f22hh9xh-poke-2.4/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>poke</li>
  </ul>
</details>

---

###### Maintainer pings

cc @AndersonTorres @kira-bruneau for [testing](https://github.com/ryantm/nixpkgs-update/blob/master/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

r-ryantm committed:
poke: 2.3 -> 2.4

viraptor reviewed (approved):
Result of `nixpkgs-review pr 182937` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>poke</li>
  </ul>
</details>

works

kira-bruneau committed:
poke: move share/vim from $lib to $out

kira-bruneau committed:
poke: only wrap poke-gui with TCLLIBPATH

kira-bruneau committed:
poke: fix guiSupport

kira-bruneau committed:
poke: fix cross-compilation

kira-bruneau commented:
I noticed a couple of issues I pushed some fixes for:
- `share/vim` was being installed to `$lib` so I updated this to move it to `$out`
- Fixed `guiSupport` (it still doesn't look like it's ready to be enabled by default)
- Fixed cross-compiliation (verified with `nix build .#pkgsCross.aarch64-multiplatform.poke`)

Does this still look ok to you @viraptor?

kira-bruneau committed:
poke: mark aarch64-darwin as a bad platform

Jitter fails to compile on aarch64-darwin:

Undefined symbols for architecture arm64:
  "_jitter_print_context_kind_destroy", referenced from:
      _jitter_print_libtextstyle_finalize in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_jitter_print_context_kind_make_trivial", referenced from:
      _jitter_print_libtextstyle_initialize in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_jitter_print_context_make", referenced from:
      _jitter_print_context_make_libtextstyle in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
     (maybe you meant: _jitter_print_context_make_libtextstyle)
  "_ostream_flush", referenced from:
      _jitter_print_context_libtextstyle_flush in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_ostream_write_mem", referenced from:
      _jitter_print_context_libtextstyle_print_chars in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_styled_ostream_begin_use_class", referenced from:
      _jitter_print_context_libtextstyle_begin_decoration in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_styled_ostream_end_use_class", referenced from:
      _jitter_print_context_libtextstyle_end_decoration in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_styled_ostream_set_hyperlink", referenced from:
      _jitter_print_context_libtextstyle_begin_decoration in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
      _jitter_print_context_libtextstyle_end_decoration in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
ld: symbol(s) not found for architecture arm64

kira-bruneau commented:
Result of `nixpkgs-review pr 182937` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>poke</li>
  </ul>
</details>

viraptor commented on pkgs/applications/editors/poke/default.nix:
```diff
@@ -85,10 +104,11 @@ in stdenv.mkDerivation rec {
     maintainers = with maintainers; [ AndersonTorres kira-bruneau ];
     platforms = platforms.unix;
-    changelog = "https://git.savannah.gnu.org/cgit/poke.git/plain/ChangeLog?h=releases/poke-${version}";
+    badPlatforms = [ "aarch64-darwin" ]; # Undefined symbols for architecture arm64
```
TIL badPlatforms. But if it's expected to work and fails building, should it be marked `broken` instead here?

kira-bruneau replied:
As far as I know `badPlatforms` is just a nicer & more declarative way to mark a package as broken on a specific platform. The only difference is if you want to force build it you need to specify `NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM` instead of `NIXPKGS_ALLOW_BROKEN`.

viraptor replied:
I mean the difference is semantic - this package is supported, but is broken on that system. (i.e. someone can attempt fix it without porting the app)

kira-bruneau replied:
From what I can tell, it doesn't look this is a Nix packaging problem and I don't think it can be fixed without patching poke. I also don't think there's really much of a semantic difference between a supported platform and a broken platform when the project doesn't mention support for `aarch64-darwin`.

AndersonTorres replied:
`broken` is a Boolean value. I treat `broken = isDarwin` as _"well, it is not working because of some Darwin quirk"_.

On the other hand `badPlatforms` is because the software is not made to build on it. An example would be run ZSNES on an ARM machine - because ZSNES is made in x86 assembly.

As far as I remember, badPlatforms is used by Hydra in order to not try tro build on those bad platforms.

kira-bruneau replied:
Ok, I've been convinced :sweat_smile:, I opened a PR for this here: #183873.

viraptor reviewed (approved):
still builds and works

jiegec commented:
Comparing https://nix-cache.s3.amazonaws.com/log/xkkkf42d9kfvlnm3a6zvh8y7infi608x-poke-2.4.drv and https://hydra.nixos.org/log/f3wm1y74lcfjg98jn7gv0alaakkpm2vi-poke-2.3.drv, it seems that `-Wl,-undefined -Wl,dynamic_lookup` is missing on aarch64-darwin:

x86_64-darwin:

```
libtool: link: clang -dynamiclib -Wl,-undefined -Wl,dynamic_lookup -o non-convenience-lib/.libs/libjitter-libtextstyle.0.dylib   -Wl,-force_load,lib/.libs/libjitter-libtextstyle.a   -g -O2   -install_name  /nix/store/4ni5kvcc4xwlhi9sskla6ssk2zmp551h-poke-2.4-lib/lib/libjitter-libtextstyle.0.dylib -compatibility_version 1 -current_version 1.0 -Wl,-single_module
```

aarch64-darwin:

```
libtool: link: clang -dynamiclib  -o non-convenience-lib/.libs/libjitter-libtextstyle.0.dylib   -Wl,-force_load,lib/.libs/libjitter-libtextstyle.a   -g -O2   -install_name  /nix/store/56i57vbykg6vfmzaic64k5p1aqs771l0-poke-2.3-lib/lib/libjitter-libtextstyle.0.dylib -compatibility_version 1 -current_version 1.0 -Wl,-single_module
Undefined symbols for architecture arm64:
  "_jitter_print_context_kind_destroy", referenced from:
      _jitter_print_libtextstyle_finalize in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_jitter_print_context_kind_make_trivial", referenced from:
      _jitter_print_libtextstyle_initialize in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_jitter_print_context_make", referenced from:
      _jitter_print_context_make_libtextstyle in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
     (maybe you meant: _jitter_print_context_make_libtextstyle)
  "_ostream_flush", referenced from:
      _jitter_print_context_libtextstyle_flush in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_ostream_write_mem", referenced from:
      _jitter_print_context_libtextstyle_print_chars in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_styled_ostream_begin_use_class", referenced from:
      _jitter_print_context_libtextstyle_begin_decoration in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_styled_ostream_end_use_class", referenced from:
      _jitter_print_context_libtextstyle_end_decoration in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
  "_styled_ostream_set_hyperlink", referenced from:
      _jitter_print_context_libtextstyle_begin_decoration in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
      _jitter_print_context_libtextstyle_end_decoration in libjitter-libtextstyle.a(jitter-print-libtextstyle.o)
ld: symbol(s) not found for architecture arm64
```

kira-bruneau commented:
:heart: Thanks for looking into this @jiegec! I don't have an M1 mac to test this on, but do you know if adding the `autoreconfHook` fixes the build then?

```
diff --git a/pkgs/applications/editors/poke/default.nix b/pkgs/applications/editors/poke/default.nix
index fba01f5..e4cb8b1 100644
--- a/pkgs/applications/editors/poke/default.nix
+++ b/pkgs/applications/editors/poke/default.nix
@@ -1,6 +1,7 @@
 { lib
 , stdenv
 , fetchurl
+, autoreconfHook
 , gettext
 , help2man
 , pkg-config
@@ -40,6 +41,7 @@ in stdenv.mkDerivation rec {
   strictDeps = true;
 
   nativeBuildInputs = [
+    autoreconfHook
     gettext
     pkg-config
     texinfo
```

jiegec commented:
@kira-bruneau Yes it works in my M1.
